deploy.sh

```diff
@@ -168,34 +272,87 @@ python3 $secret_script --apiToken=$apiToken \
 --password=$DOCKER_PASSWORD \
 --force
 
+# Format name for kubernetes 
+HELM_BINDERHUB_NAME=$(echo ${BINDERHUB_NAME} | tr -cd '[:alnum:]' | tr '[:upper:]' '[:lower:]' | sed -re 's/^([.-]+)//' -re 's/([.-]+)$//' )
+
 echo "--> Installing Helm chart"
 helm install jupyterhub/binderhub \
 --version=$BINDERHUB_VERSION \
---name=$BINDERHUB_NAME \
---namespace=$BINDERHUB_NAME \
+--name=$HELM_BINDERHUB_NAME \
+--namespace=$HELM_BINDERHUB_NAME \
 -f ./secret.yaml \
 -f ./config.yaml \
---timeout=3600
+--timeout=3600 | tee helm-chart-install.log
 
 # Wait for  JupyterHub, grab its IP address, and update BinderHub to link together:
-echo "--> Retrieving BinderHub IP"
-jupyterhub_ip=`kubectl --namespace=$BINDERHUB_NAME get svc proxy-public | awk '{ print $4}' | tail -n 1`
+echo "--> Retrieving JupyterHub IP"
+jupyterhub_ip=`kubectl --namespace=$HELM_BINDERHUB_NAME get svc proxy-public | awk '{ print $4}' | tail -n 1`
 while [ "$jupyterhub_ip" = '<pending>' ] || [ "$jupyterhub_ip" = "" ]
 do
-    echo "JupyterHub IP: $jupyterhub_ip"
-    sleep 5
-    jupyterhub_ip=`kubectl --namespace=$BINDERHUB_NAME get svc proxy-public | awk '{ print $4}' | tail -n 1`
+    echo "Sleeping 30s before checking again"
+    sleep 30
+    jupyterhub_ip=`kubectl --namespace=$HELM_BINDERHUB_NAME get svc proxy-public | awk '{ print $4}' | tail -n 1`
+    echo "JupyterHub IP: $jupyterhub_ip" | tee jupyterhub-ip.log
 done
 
 echo "--> Finalising configurations"
-python3 $config_script -id=$DOCKER_USERNAME \
---prefix=$DOCKER_IMAGE_PREFIX \
--org=$DOCKER_ORGANISATION \
---jupyterhub_ip=$jupyterhub_ip \
---force
+if [ -z "$DOCKER_ORGANISATION" ] ; then
+  python3 $config_script -id=$DOCKER_USERNAME \
+  --prefix=$DOCKER_IMAGE_PREFIX \
+  --jupyterhub_ip=$jupyterhub_ip \
+  --force
+else
+  python3 $config_script -id=$DOCKER_USERNAME \
+  --prefix=$DOCKER_IMAGE_PREFIX \
+  -org=$DOCKER_ORGANISATION \
+  --jupyterhub_ip=$jupyterhub_ip \
+  --force
+fi
 
+helm repo update
 echo "--> Updating Helm chart"
-helm upgrade $BINDERHUB_NAME jupyterhub/binderhub \
---version=$BINDERHUB_VERSION \
+helm upgrade $HELM_BINDERHUB_NAME jupyterhub/binderhub \
+--version=$HELM_BINDERHUB_VERSION \
 -f secret.yaml \
--f config.yaml
+-f config.yaml | tee helm-upgrade.log
+
+# Print Binder IP address
+binder_ip=`kubectl --namespace=$HELM_BINDERHUB_NAME get svc binder | awk '{ print $4}' | tail -n 1`
+while [ "$binder_ip" = '<pending>' ] || [ "$binder_ip" = "" ]
+do
+    echo "Sleeping 30s before checking again"
+    sleep 30
+    binder_ip=`kubectl --namespace=$HELM_BINDERHUB_NAME get svc binder | awk '{ print $4}' | tail -n 1`
+    echo "Binder IP: $binder_ip" | tee binder-ip.log
+done
+
+if [ ! -z $BINDERHUB_CONTAINER_MODE ] ; then
+  # Finally, save outputs to blob storage
+  # 
+  # Create a storage account
+  echo "--> Creating storage account"
+  CONTAINER_NAME="${CONTAINER_NAME}"
```

Comment:
@tmbgreaves Where does this variable get set? I think this is why the pushing of the logs in still failing.

Reply:
```suggestion
  CONTAINER_NAME="${BINDERHUB_NAME}setuplogs"
```
Might work?